variables.tf

```diff
@@ -369,95 +369,109 @@ variable "custom_security_group_ids" {
-  type        = string
-  default     = null
+variable "logs_agent_exclude_log_source_paths" {
+  type        = list(string)
+  description = "The list of log sources to exclude. Specify the paths that the Logs agent ignores."
+  default     = []
+  nullable    = false
 }
 
-variable "log_analysis_agent_custom_line_exclusion" {
-  description = "Log Analysis agent custom configuration for line exclusion setting LOGDNA_K8S_METADATA_LINE_EXCLUSION. See https://github.com/logdna/logdna-agent-v2/blob/master/docs/KUBERNETES.md#configuration-for-kubernetes-metadata-filtering for more info."
-  type        = string
-  default     = null
+variable "logs_agent_selected_log_source_paths" {
+  type        = list(string)
+  description = "The list of specific log sources paths. Logs will only be collected from the specified log source paths. If no paths are specified, it will send logs from `/var/log/containers`."
+  default     = []
+  nullable    = false
 }
 
-variable "log_analysis_agent_name" {
-  description = "Log Analysis agent name. Used for naming all kubernetes and helm resources on the cluster."
-  type        = string
-  default     = "logdna-agent"
+variable "logs_agent_log_source_namespaces" {
+  type        = list(string)
+  description = "The list of namespaces from which logs should be forwarded by agent. If namespaces are not listed, logs from all namespaces will be sent."
+  default     = []
   nullable    = false
 }
 
-variable "log_analysis_agent_namespace" {
+variable "logs_agent_iam_mode" {
   type        = string
-  description = "Namespace where to deploy the Log Analysis agent. Default value is 'ibm-observe'"
-  default     = "ibm-observe"
-  nullable    = false
+  default     = "TrustedProfile"
+  description = "IAM authentication mode: `TrustedProfile` or `IAMAPIKey`."
 }
 
-variable "log_analysis_agent_tolerations" {
-  description = "List of tolerations to apply to Log Analysis agent."
+variable "logs_agent_iam_environment" {
+  type        = string
+  default     = "PrivateProduction"
+  description = "IAM authentication Environment: `Production` or `PrivateProduction` or `Staging` or `PrivateStaging`. `Production` specifies the public endpoint & `PrivateProduction` specifies the private endpoint."
+}
+
+variable "logs_agent_additional_metadata" {
+  description = "The list of additional metadata fields to add to the routed logs."
   type = list(object({
-    key               = optional(string)
-    operator          = optional(string)
-    value             = optional(string)
-    effect            = optional(string)
-    tolerationSeconds = optional(number)
+    key   = optional(string)
+    value = optional(string)
   }))
-  default = [{
-    operator = "Exists"
-  }]
+  default = []
+}
+
+variable "cloud_logs_ingress_endpoint" {
+  description = "The host for IBM Cloud Logs ingestion. Ensure you use the ingress endpoint. See https://cloud.ibm.com/docs/cloud-logs?topic=cloud-logs-endpoints_ingress."
```

Comment:
I guess this should say required if `logs_agent_enabled` is set to true. And add validation for this too

Reply:
<img width="1247" alt="Screenshot 2024-10-21 at 4 06 30 PM" src="https://github.com/user-attachments/assets/fba8c709-f064-4d4f-9668-32e08081efcc">
There is already a validation present in the obs agent module.